server/src/main/java/com/google/finapp/SpannerDaoImpl.java

```diff
@@ -114,13 +110,8 @@ public void createCustomerRole(
   @Override
   public ImmutableMap<ByteArray, BigDecimal> moveAccountBalance(
       ByteArray fromAccountId, ByteArray toAccountId, BigDecimal amount)
```

Comment:
I see you catch negative `amount` in the caller now, but this is still a public method. Should negative `amount`s be legal here?

Reply:
This is a non-public class, so it is ok-ish to assume things about its input. This is also documented in the interface were we mention that the amount is expected to be non-negative. I do see the concern, but I think in this case it's an internal implementation class and there's documentation in the interface so it's probably ok.